server/src/rooms/game/commands/index.ts

```diff
@@ -215,41 +219,44 @@ export class SetNextPhaseCmd implements Command {
       case Phase.events: {
         const events = [];
         if (this.game.state.currentEvent) {
-          events.push(new MarsEventFinalized())
+          events.push(new MarsEventFinalized());
         }
         if (
           this.game.state.marsEventsProcessed + 1 >=
           this.game.state.marsEvents.length
         ) {
           events.push(new EnteredInvestmentPhase());
-          this.gameOver(this.upkeep);
+          console.log(this.game.state.upkeep);
+          
         } else {
           events.push(new ReenteredMarsEventPhase());
           events.push(new MarsEventInitialized());
-          this.gameOver(this.upkeep);
+          
         }
 
+        events.push(new EnteredDefeatPhase());
         return events;
       }
       case Phase.invest:
         return [new EnteredTradePhase()];
       case Phase.trade:
         return [new EnteredPurchasePhase()];
       case Phase.purchase:
-        this.gameOver(this.upkeep);
-        return [new EnteredDiscardPhase()];
+        
+        
+        return [new EnteredDiscardPhase(), new EnteredDefeatPhase()];
```

Comment:
can you explain what's going on here? why are we adding the `EnteredDefeatPhase` here?